src/Common/src/CoreLib/System/Memory.cs

```diff
@@ -71,21 +77,23 @@ public Memory(T[] array, int start, int length)
             if (default(T) == null && array.GetType() != typeof(T[]))
                 ThrowHelper.ThrowArrayTypeMismatchException();
             if ((uint)start > (uint)array.Length || (uint)length > (uint)(array.Length - start))
-                ThrowHelper.ThrowArgumentOutOfRangeException();
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
 
             _object = array;
             _index = start;
             _length = length;
         }
-        
+
         // Constructor for internal use only.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal Memory(OwnedMemory<T> owner, int index, int length)
         {
             if (owner == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.ownedMemory);
-            if (index < 0 || length < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
+            if (index < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
```

Comment:
This internal constructor and the callers are validating the arguments already. Do we need to validate the arguments here again?

These constructors are hot paths, with AggresiveInline. Any unnecessary cruft here impacts microbencmarks; and has multiplicative effect on code size.

Reply:
Context for the checks being there:
https://github.com/dotnet/coreclr/pull/13583/files/f538e879ed80d82c155270796d2ab20bdd63dbe5#r135248195

Reply:
I am seeing there is no strong opinion to keep the checks even in the comment that @ahsonkhan pointed at. do you agree with removing the checks (for index and length)?

Reply:
It is odd to have a partial argument validation. I think we should do no validation here, and let the callers take care of it as necessary.